src/mount/mount-tool.c

```diff
@@ -384,19 +384,34 @@ static int parse_argv(int argc, char *argv[]) {
                         if (!u)
                                 return log_oom();
 
-                        r = chase(u, NULL, 0, &arg_mount_what, NULL);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to make path %s absolute: %m", u);
+                        if (streq(u, "tmpfs")) {
```

Comment:
this creates am ambiguity. what happens if you have a file by that name in the local dir?

Reply:
Relative paths are already not supported by systemd-mount, they have to be absolute. And it works as expected, just like calling mount:

```
$ cd /tmp
$ echo foo > tmpfs
$ sudo mount tmpfs -t tmpfs /tmp/img
$ ls /tmp/img/ -lha
total 0
drwxrwxrwt  2 root root   40 Aug 16 11:16 .
drwxrwxrwt 43 root root 1.9K Aug 16 11:16 ..
$ sudo umount img
$ sudo systemd-mount tmpfs /tmp/img
Started unit tmp-img.mount for mount point: /tmp/img
$ ls -la /tmp/img
total 0
drwxrwxrwt  2 root root   40 Aug 16 11:17 .
drwxrwxrwt 43 root root 1880 Aug 16 11:17 ..
```

Reply:
> > Relative paths are already not supported by systemd-mount, they have to be absolute.
> 
> They have to? That's news to me. Works fine here.

Right, I was looking at the path_is_absolute() check but it's for the remote case, never mind

> > And it works as expected, just like calling mount:
> 
> No, that example just shows nothing: you have no local file called "tmpfs", so what's that supposed to illustrate.

There is a local file called `tmpfs`

> This really shouldn't add an ambiguity, and it does. Just try it out! create a local file that contains a valid fs...

You didn't say "valid fs" before ;-)

Reply:
Ok, I've changed this to instead only parse a single argument (as mount point) in case Type=tmpfs is specified. PTAL.